frontend/components/customizer/query-string-builder.tsx

```diff
@@ -176,43 +228,59 @@ export default function QueryStringBuilder({
   exampleParams,
   initialStyle = 'flat',
   onChange,
+}: {
+  exampleParams: { [k: string]: string }
+  initialStyle?: string
+  onChange: ({
+    queryString,
+    isComplete,
+  }: {
+    queryString: string
+    isComplete: boolean
+  }) => void
 }) {
-  const [queryParams, setQueryParams] = useState(() => {
+  const [queryParams, setQueryParams] = useState(() =>
     // For each of the custom query params defined in `exampleParams`,
     // create empty values in `queryParams`.
-    const result = {}
-    Object.entries(exampleParams).forEach(([name, value]) => {
-      // Custom query params are either string or boolean. Inspect the example
-      // value to infer which one, and set empty values accordingly.
-      // Throughout the component, these two types are supported in the same
-      // manner: by inspecting this value type.
-      const isStringParam = typeof value === 'string'
-      result[name] = isStringParam ? value : true
-    })
-    return result
-  })
+    Object.entries(exampleParams).reduce(
+      (accum, [name, value]) => {
+        // Custom query params are either string or boolean. Inspect the example
+        // value to infer which one, and set empty values accordingly.
+        // Throughout the component, these two types are supported in the same
+        // manner: by inspecting this value type.
+        const isStringParam = typeof value === 'string'
+        accum[name] = isStringParam ? '' : true
+        return accum
+      },
+      {} as { [k: string]: string | boolean }
+    )
+  )
   // For each of the standard badge options, create empty values in
   // `badgeOptions`. When `initialStyle` has been provided, use it.
   const [badgeOptions, setBadgeOptions] = useState(() =>
-    supportedBadgeOptions.reduce((accum, { name }) => {
-      if (name === 'style') {
-        accum[name] = initialStyle
-      } else {
-        accum[name] = ''
-      }
-      return accum
-    }, {})
+    supportedBadgeOptions.reduce(
+      (accum, { name }) => {
+        if (name === 'style') {
+          accum[name] = initialStyle
+        } else {
+          accum[name] = ''
+        }
+        return accum
+      },
+      {} as Record<BadgeOptionName, string>
+    )
   )
 
-  function handleServiceQueryParamChange(event) {
-    const { name, type } = event.target
-    const value =
-      type === 'checkbox' ? event.target.checked : event.target.value
-    setQueryParams({ ...queryParams, [name]: value })
+  function handleServiceQueryParamChange({
+    target: { name, type, checked, value },
```

Comment:
Any potential issues we could run into here since `type` is a reserved/key word in TypeScript?

Reply:
Will update that, though if that really is a problem, wonder if we should open a new issue to have the linter check for it.

Reply:
Everything was working correctly as far as I could tell but it wouldn't surprise me if at some point the TypeScript compiler would start to complain